content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,388 @@
+# Checkpoints
+
+ML checkpoints are an important part of deep learning because ML engineers like
+to save the model files at certain points during a training process.
+
+With DVC experiments and checkpoints, you can:
+
+- Implement the best practice in deep learning to save your model weights as
+  checkpoints.
+- Add reproducibility to the messy deep learning process.
+- Track all code and data changes corresponded to the checkpoints.
+- Return a report when metrics start diverging.
+
+You can also automate the process of tracking every training epoch, return to
+any previous checkpoint, see when metrics converge or when the learning rate
+needs to be adjusted, and even recover model weights, parameters, and code. When
+you adjust parameters and code, DVC tracks those changes for you without adding
+a lot of commits to your workspace.
+
+[The way checkpoints are implemented by DVC](/blog/experiment-refs) utilizes
+_ephemeral_ experiment commits and experiment branches within DVC. They are
+created using the metadata from experiments and are tracked with the `exps`
+custom Git reference.
+
+You can add experiments to your Git history by committing the experiment you
+want to track, which you'll see later in this tutorial.
+
+## Setting up the project
+
+This tutorial is going to cover how to implement checkpoints in a ML project
+using DVC. We're going to train a model to identify handwritten digits based on
+the MNIST dataset.
+
+You can follow along with the steps here or you can clone the repo directly from
+GitHub and play with it. To clone the repo, run the following commands.
+
+```bash
+git clone https://github.com/iterative/checkpoints-tutorial
+cd checkpoints-tutorial
+```
+
+It is highly recommended you create a virtual environment for this example. You
+can do that by running: `python3 -m venv .venv`.
+
+Once your virtual environment is installed, you can start it up with one of the
+following commands.
+
+- On Mac/Linux: `source .venv/bin/activate`
+- On Windows: `.\.venv\Scripts\activate`
+
+Once you have your environment set up, you can install the dependencies by
+running: `pip install -r requirements.txt`. This will download all of the
+packages you need to run the example. Now you have everything you need to get
+started with experiments and checkpoints.
+
+## Setting up a DVC pipeline
+
+If you don't have a DVC pipeline setup in your project, adding one only takes a
+few commands. At the root of the project, run `dvc init`. This sets up the files
+you need for your DVC pipeline to work.
+
+Next we'll add a stage to our pipeline to run our download the data we need for
+training our model. In your terminal, run
+`dvc stage add -n download -d download.py -o data/MNIST python download.py`.
```

Comment:
This command is not needed since `train.py` does the downloading. So, you can significantly simplify this section and combine it with the next one "Enabling DVC pipelines for checkpoint experiments".

Only two simple code blocks are needed to build the entire DVC pipeline:
```bash
$ dvc init
$ dvc stage add -n train -d data/MNIST -d train.py -c model.pt --plots-no-cache predictions.json -p seed,lr,weight_decay --live dvclive python train.py
``` 
and 
```
$ git add .
$ git commit -m 'DVC pipeline'
```

One of the goals of this section should be to show how easy to make a DVC project with checkpoints from a regular ML repository.